### PR TITLE
Cloud 781

### PR DESCRIPTION
**Pull request by lpuskas**
* Moved bootstrap related code from the core.flow package to a dedicated place
* Moved service implementations down to the service package (from the flow package) kept only the high level classes there
* Extracted inner factory class to loosen dependencies
* Improved some messaging

@martonsereg @keyki comment it, and merge it please ;)

### CODE REVIEW

**Comment by martonsereg** [comment]
@lpuskas please specify only 1 or at most 2 people to review it, because no one feel like it is his duty

**Comment by martonsereg** [comment]
LGTM

File: core/src/test/java/com/sequenceiq/cloudbreak/core/flow/SimpleTransitionKeyServiceTest.java [hunk 1]
```diff
@@ -6,6 +6,8 @@
 import org.slf4j.Logger;
```
Comment (schfeca75) [inline core/src/test/java/com/sequenceiq/cloudbreak/core/flow/SimpleTransitionKeyServiceTest.java]: the class should be moved to the ...flow.service package

File: core/src/test/java/com/sequenceiq/cloudbreak/core/flow/ReactorFlowManagerTest.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
 import com.sequenceiq.cloudbreak.conf.TestConfig;
```
Comment (schfeca75) [inline core/src/test/java/com/sequenceiq/cloudbreak/core/flow/ReactorFlowManagerTest.java]: the class should be moved to the ...flow.service package

**Pull request merged**